code/game/objects/structures/transit_tubes/transit_tube_construction.dm

```diff
@@ -4,58 +4,46 @@
 /obj/structure/c_transit_tube
 	name = "unattached transit tube"
 	icon = 'icons/obj/atmospherics/pipes/transit_tube.dmi'
-	icon_state = "E-W" //icon_state decides which tube will be built
+	icon_state = "straight" //icon_state decides which tube will be built
 	density = 0
-	layer = ABOVE_OBJ_LAYER //same as the built tube
+	layer = LOW_ITEM_LAYER //same as the built tube
 	anchored = 0
+	var/flipped = 0
+	var/build_type = /obj/structure/transit_tube
+	var/flipped_build_type
+	var/base_icon
 
 /obj/structure/c_transit_tube/examine(mob/user)
 	..()
 	user << "<span class='notice'>Alt-click to rotate it clockwise.</span>"
 
-//wrapper for turn that changes the transit tube formatted icon_state instead of the dir
-/obj/structure/c_transit_tube/proc/tube_turn(angle)
-	var/list/badtubes = list("W-E", "W-E-Pass", "S-N", "S-N-Pass", "SW-NE", "SE-NW")
-	var/list/split_text = splittext(icon_state, "-")
-	for(var/i=1; i<=split_text.len; i++)
-		var/curdir = text2dir_extended(split_text[i]) //0 if not a valid direction (e.g. Pass, Block)
-		if(curdir)
-			split_text[i] = dir2text_short(turn(curdir, angle))
-	var/newdir = jointext(split_text, "-")
-	if(badtubes.Find(newdir))
-		split_text.Swap(1,2)
-		newdir = jointext(split_text, "-")
-	icon_state = newdir
+/obj/structure/c_transit_tube/proc/tube_rotate()
+	setDir(turn(dir, -90))
 
 /obj/structure/c_transit_tube/proc/tube_flip()
-	var/list/split_text = splittext(icon_state, "-")
-	//skip straight pipes
-	if(length(split_text[2]) < 2)
-		return
-	//for junctions, just swap the diagonals with each other
-	if(split_text.len == 3 && split_text[3] != "Pass")
-		split_text.Swap(2,3)
-	else if(length(split_text[1]) == 2 && length(split_text[2]) == 2) //diagonals
-		split_text[1] = copytext(split_text[1],1,2) + copytext(split_text[2],2,3)
-		split_text[2] = copytext(split_text[2],1,2) + ((copytext(split_text[2],2,3) == "E") ? "W" : "E")
-	//for curves, swap the diagonal direction that is not in the same axis as the cardinal direction
-	else
-		if(split_text[1] == "N" || split_text[1] == "S")
-			split_text[2] = copytext(split_text[2],1,2) + ((copytext(split_text[2],2,3) == "E") ? "W" : "E")
+	if(flipped_build_type)
+		flipped = !flipped
+		var/cur_flip = flipped
+		if(initial(flipped))
+			cur_flip = !cur_flip

```

Comment:
could condense these three lines into `var/cur_flip = initial(flipped) ? !flipped : flipped` but it's not a big deal